### PR TITLE
SSO Core - OIDC + OAuth2.0 docs

### PR DESCRIPTION
**Pull request by imilinovic**
### Description

Add docs for the new SSO OIDC module for Enterprise.

### Pull request type

Please check what kind of PR this is:

- [ ] Fix or improvement of an existing page
- [x] New documentation page, release related

### Related PRs and issues

PR this doc page is related to:
[PR2045](https://github.com/memgraph/memgraph/pull/2045)
(especially necessary if the PR is related to a release)

Closes:
(paste the link to the issue it closes)

### Checklist:

- [x] Check all content with Grammarly
- [x] Perform a self-review of my code
- [x] Make corresponding changes to the rest of the documentation (consult with the DX team)
- [x] The build passes locally
- [x] My changes generate no new warnings or errors
- [x] Add a corresponding label
- [x] If release-related, add a product and version label
- [x] If release-related, add release note on product PR

### CODE REVIEW

File: pages/database-management/authentication-and-authorization/auth-system-integrations.mdx [hunk 1]
```diff
@@ -409,3 +409,80 @@ allowed to log in because her role (moderator) already exists. Carol and Dave
...
+For using OIDC with Microsoft Entra ID you have to set the following environmental variables:
+
+```bash
+MEMGRAPH_SSO_ENTRA_ID_OIDC_TENANT_ID=
+MEMGRAPH_SSO_ENTRA_ID_OIDC_CLIENT_ID=
+MEMGRAPH_SSO_ENTRA_ID_OIDC_ROLE_MAPPING=
+MEMGRAPH_SSO_ENTRA_ID_OIDC_USERNAME=
+```
+
+You can find the client and tenant ID in the app overview in Azure. For a more detailed explanation see [this](/pages/data-visualization/user-manual/single-sign-on.mdx).
```
Comment (hal-eisen-MG) [inline pages/database-management/authentication-and-authorization/auth-system-integrations.mdx:448]: Above, you have the tenant variable before the client variable. But here you have the reverse ordering. Please be consistent.

Comment (hal-eisen-MG) [reply pages/database-management/authentication-and-authorization/auth-system-integrations.mdx:448]: Where can the reader find the role mapping? What about the username?

File: pages/database-management/authentication-and-authorization/auth-system-integrations.mdx [hunk 2]
```diff
@@ -409,3 +409,80 @@ allowed to log in because her role (moderator) already exists. Carol and Dave
...
+
+##### Role mapping
+Authorization is done by mapping the Entra ID roles from the roles field in the access token to the Memgraph roles. Make sure that each user only has one role on Entra ID.
```
Comment (hal-eisen-MG) [inline pages/database-management/authentication-and-authorization/auth-system-integrations.mdx:451]: Explain how the colon (:) and semi-colon (;) work here.

File: pages/database-management/authentication-and-authorization/auth-system-integrations.mdx [hunk 3]
```diff
@@ -409,3 +409,80 @@ allowed to log in because her role (moderator) already exists. Carol and Dave
...
+One example of role mapping is `entra.admin:memadmin; entra.user:memuser`.
+
+##### Username
+The username variable tells the OIDC module what to use as the username. It has the format `token-type:field`. 
+Token type can be `id` or `access` depending on whether you want to use a field from the access or the ID token for the username.
```
Comment (hal-eisen-MG) [inline pages/database-management/authentication-and-authorization/auth-system-integrations.mdx:456]: Same note as above. Be consistent in the ordering of talking about things.

File: pages/database-management/authentication-and-authorization/auth-system-integrations.mdx [hunk 4]
```diff
@@ -409,3 +409,80 @@ allowed to log in because her role (moderator) already exists. Carol and Dave
...
+The module is written in Python 3 and requires the following libraries:
+ - `jwt` - used to validate tokens.
+ - `requests` - used to retrieve public keys for validating tokens.
+
+The required packages are already installed in the docker image or native Memgraph if you followed the installation procedure correctly.
+
+### Module configuration
+
+The module is located at `/etc/memgraph/auth_module/oidc.py`.
+The module has no configuration file. You only have to run Memgraph with some environmental variables.
```
Comment (hal-eisen-MG) [inline pages/database-management/authentication-and-authorization/auth-system-integrations.mdx:435]: Remove "only"

What happens if the user provides BOTH entra and okta variables?

File: pages/database-management/authentication-and-authorization/auth-system-integrations.mdx (hunk 3 above)
Comment (hal-eisen-MG) [reply pages/database-management/authentication-and-authorization/auth-system-integrations.mdx:456]: Please explain more about the difference between "id" and "access". Perhaps link to an Okta page?

File: pages/database-management/authentication-and-authorization/auth-system-integrations.mdx [hunk 5]
```diff
@@ -409,3 +409,80 @@ allowed to log in because her role (moderator) already exists. Carol and Dave
...
+MEMGRAPH_SSO_OKTA_OIDC_CLIENT_ID=
+MEMGRAPH_SSO_OKTA_OIDC_AUTHORIZATION_SERVER=api://default
+MEMGRAPH_SSO_OKTA_OIDC_ROLE_MAPPING=
+MEMGRAPH_SSO_OKTA_OIDC_USERNAME=
+```
+
+Issuer is `https://{your-okta-domain}.okta.com/oauth2/default/`. You can find the client ID on the Admin panel -> Applications -> General. You can find the authorization server on the Admin panel -> Security -> API -> Authorization Servers -> Audience. By default, it is set to `api://default`.
+
+##### Role mapping
+Authorization is done by mapping the Okta groups from the `groups` field in the access token to the Memgraph roles. Make sure that each user only has one role on Entra ID.
```
Comment (hal-eisen-MG) [inline pages/database-management/authentication-and-authorization/auth-system-integrations.mdx:474]: Why are we talking about Entra when we're in the Okta section?

**Review by kgolubic — APPROVED** [review]
Everything looks OK to me.

**Review by hal-eisen-MG — APPROVED** [review]
LGTM

**Pull request merged**